python/knowhere/knowhere.i

```diff
@@ -94,73 +96,97 @@ DOWNCAST ( IndexAnnoy )
 %apply (float *IN_ARRAY1, int DIM1) {(float *dis, int len)}
 %apply (float* INPLACE_ARRAY2, int DIM1, int DIM2){(float *dis,int nq_1,int k_1)}
 %apply (int *INPLACE_ARRAY2, int DIM1, int DIM2){(int *ids,int nq_2,int k_2)}
+
+
 %inline %{
 
 
-knowhere::DatasetPtr ArrayToDataSet( float* xb,int nb, int dim){
+knowhere::DatasetPtr
+ArrayToDataSet(float* xb, int nb, int dim) {
     auto ret_ds = std::make_shared<Dataset>();
     ret_ds->Set<int64_t>("rows", nb);
     ret_ds->Set<int64_t>("dim", dim);
-    ret_ds->Set<const void *>("tensor", xb);
+    ret_ds->Set<const void*>("tensor", xb);
     return ret_ds;
 };
 
-void DumpResultDataSet(knowhere::DatasetPtr result, float *dis, int nq_1, int k_1, 
-                       int *ids,int nq_2, int k_2){
-    auto ids_ = result->Get<const int64_t *>("ids");
-    auto dist_ = result->Get<const float *>("distance");
-    assert(nq_1==nq_2);
-    assert(k_1==k_2);
+void
+DumpResultDataSet(const knowhere::DatasetPtr& result, float* dis, int nq_1, int k_1, int* ids, int nq_2, int k_2) {
+    auto ids_ = result->Get<const int64_t*>("ids");
```

Comment:
suggest to use GetDatasetIDs() and GetDatasetDistance()

Reply:
reduce c++ head api export in python.